tests/lib/smf/src/test_lib_self_transition_smf.c

```diff
@@ -9,89 +10,104 @@
-#define STATE_B_ENTRY_BIT       (1 << 5)
-
-/* Run 1 */
-#define STATE_B_RUN_BIT         (1 << 6)
-#define STATE_B_EXIT_BIT        (1 << 7)
-#define PARENT_AB_EXIT_BIT      (1 << 8)
-#define PARENT_C_ENTRY_BIT      (1 << 9)
-#define STATE_C_ENTRY_BIT       (1 << 10)
-
-/* Run 2 */
-#define STATE_C_1ST_RUN_BIT     (1 << 11)
-
-/* Run 3 */
-#define STATE_C_2ND_RUN_BIT     (1 << 12)
-#define PARENT_C_RUN_BIT        (1 << 13)
-#define STATE_C_EXIT_BIT        (1 << 14)
-#define PARENT_C_EXIT_BIT       (1 << 15)
-
-#define TEST_PARENT_ENTRY_VALUE_NUM 0
-#define TEST_PARENT_RUN_VALUE_NUM   3
-#define TEST_PARENT_EXIT_VALUE_NUM  8
-#define TEST_ENTRY_VALUE_NUM        1
+#define SMF_RUN 5
+
+/* Initial Setup: Testing initial transitions */
+#define ROOT_ENTRY_BIT      BIT(0)
+#define PARENT_AB_ENTRY_BIT BIT(1)
+#define STATE_A_ENTRY_BIT   BIT(2)
+
+/* Run 0: normal state transition */
+#define STATE_A_RUN_BIT   BIT(3)
+#define STATE_A_EXIT_BIT  BIT(4)
+#define STATE_B_ENTRY_BIT BIT(5)
+
+/* Run 1: Test smf_set_handled() */
+#define STATE_B_1ST_RUN_BIT BIT(6)
+
+/* Run 2: Normal state transition via parent */
+#define STATE_B_2ND_RUN_BIT    BIT(7)
+#define PARENT_AB_RUN_BIT      BIT(8)
+#define STATE_B_EXIT_BIT       BIT(9)
+#define PARENT_AB_EXIT_BIT     BIT(10)
+#define PARENT_C_1ST_ENTRY_BIT BIT(11)
+#define STATE_C_1ST_ENTRY_BIT  BIT(12)
+
+/* Run 3: PARENT_C executes transition to self  */
+#define STATE_C_1ST_RUN_BIT    BIT(13)
+#define PARENT_C_RUN_BIT       BIT(14)
+#define STATE_C_1ST_EXIT_BIT   BIT(15)
+#define PARENT_C_1ST_EXIT_BIT  BIT(16)
+#define PARENT_C_2ND_ENTRY_BIT BIT(17)
+#define STATE_C_2ND_ENTRY_BIT  BIT(18)
+
+/* Run 4: Test transition from parent state */
+#define STATE_C_2ND_RUN_BIT   BIT(19)
+#define STATE_C_2ND_EXIT_BIT  BIT(20)
+#define PARENT_C_2ND_EXIT_BIT BIT(21)
+
+/* Unused functions: Error checks if set */
+#define ROOT_RUN_BIT  BIT(22)
+#define ROOT_EXIT_BIT BIT(23)
+
+/* Number of state transitions for each test: */
+#define TEST_VALUE_NUM              22
+#define TEST_PARENT_ENTRY_VALUE_NUM 1
+#define TEST_PARENT_RUN_VALUE_NUM   8
+#define TEST_PARENT_EXIT_VALUE_NUM  10
+#define TEST_ENTRY_VALUE_NUM        2
 #define TEST_RUN_VALUE_NUM          6
-#define TEST_EXIT_VALUE_NUM         14
-#define TEST_VALUE_NUM              16
+#define TEST_EXIT_VALUE_NUM         15
+
 static uint32_t test_value[] = {
-	0x00,   /* PARENT_AB_ENTRY */
-	0x01,   /* STATE_A_ENTRY */
-	0x03,   /* STATE_A_RUN */
-	0x07,   /* PARENT_AB_RUN */
-	0x0f,   /* STATE_A_EXIT */
-	0x1f,   /* STATE_B_ENTRY */
-	0x3f,   /* STATE_B_RUN */
-	0x7f,   /* STATE_B_EXIT */
-	0xff,   /* PARENT_AB_EXIT */
-	0x1ff,  /* PARENT_C_ENTRY */
-	0x3ff,  /* STATE_C_ENTRY */
-	0x7ff,  /* STATE_C_1ST_RUN */
-	0xfff,  /* STATE_C_2ND_RUN */
-	0x1fff, /* STATE_C_EXIT */
-	0x3fff, /* PARENT_C_RUN */
-	0x7fff, /* PARENT_C_EXIT */
-	0xffff, /* FINAL VALUE */
+	/* Initial Setup */
+	0x000000, /* ROOT_ENTRY_BIT */
+	0x000001, /* PARENT_AB_ENTRY_BIT */
```

Comment:
You could use `BIT_MASK(1)` here and for the other definitions.

Reply:
I think I swapped from `BIT()` to make it easier to determine which flags had been set when I was printing it out.